Lib/test/test_configparser.py

```diff
@@ -980,6 +980,51 @@ def test_set_nonstring_types(self):
         self.assertRaises(TypeError, cf.set, "sect", 123, "invalid opt name!")
         self.assertRaises(TypeError, cf.add_section, 123)
 
+    def test_str(self):
+        self.maxDiff = None
+        cf = self.config_class(allow_no_value=True, delimiters=('=',), strict=True)
+        cf.add_section("sect1")
+        cf.add_section("sect2")
+        cf.set("sect1", "option1", "foo")
+        cf.set("sect2", "option2", "bar")
+
+        expected_str = (
+            "<ConfigParser: {'sect1': {'option1': 'foo'}, 'sect2': {'option2': 'bar'}}>"
+        )
+        self.assertEqual(str(cf), expected_str)
+
+    def test_repr(self):
+        self.maxDiff = None
+        cf = self.config_class(allow_no_value=True, delimiters=('=',), strict=True)
+        cf.add_section("sect1")
+        cf.add_section("sect2")
+        cf.add_section("sect3")
+        cf.add_section("sect4")
+        cf.add_section("sect5")
+        cf.add_section("sect6")
+        cf.set("sect1", "option1", "foo")
+        cf.set("sect2", "option2", "bar")
+        cf.read_string("")
```

Comment:
Is this to have `'loaded_sources'== ['<string'>]`? Maybe add a comment saying why we read this?

Reply:
Done